core/trino-main/src/main/java/io/trino/execution/scheduler/BinPackingNodeAllocatorService.java

```diff
@@ -206,6 +207,10 @@ void refreshNodePoolMemoryInfos()
     @VisibleForTesting
     synchronized void processPendingAcquires()
     {
+        // process EAGER_SPECULATIVE first; it will not cause deadlock even though STANDARD acquires are processed later.
```

Comment:
does this require a more detailed explanation?

Reply:
reworded